src/@chakra-ui/gatsby-plugin/components/components.utils.ts

```diff
@@ -0,0 +1,72 @@
+import { SystemStyleObject, theme } from "@chakra-ui/react"
+import { merge } from "lodash"
+
+const {
+  Accordion: accordionDefaultTheme,
+  Avatar: avatarDefaultTheme,
+  Badge: badgeDefaultTheme,
+  Breadcrumb: breadcrumbDefaultTheme,
+  Button: buttonDefaultTheme,
+  Checkbox: checkboxDefaultTheme,
+  CloseButton: closeButtonDefaultTheme,
+  Code: codeDefaultTheme,
+  Divider: dividerDefaultTheme,
+  Drawer: drawerDefaultTheme,
+  Form: formDefaultTheme,
+  FormLabel: formLabelDefaultTheme,
+  Heading: headingDefaultTheme,
+  Input: inputDefaultTheme,
+  Link: linkDefaultTheme,
+  List: listDefaultTheme,
+  Menu: menuDefaultTheme,
+  Modal: modalDefaultTheme,
+  Select: selectDefaultTheme,
+  Spinner: spinnerDefaultTheme,
+  Switch: switchDefaultTheme,
+  Table: tableDefaultTheme,
+  Tabs: tabsDefaultTheme,
+  Tag: tagDefaultTheme,
+} = theme.components
+
+export {
+  accordionDefaultTheme,
+  avatarDefaultTheme,
+  badgeDefaultTheme,
+  breadcrumbDefaultTheme,
+  buttonDefaultTheme,
+  checkboxDefaultTheme,
+  closeButtonDefaultTheme,
+  codeDefaultTheme,
+  dividerDefaultTheme,
+  drawerDefaultTheme,
+  formDefaultTheme,
+  formLabelDefaultTheme,
+  headingDefaultTheme,
+  inputDefaultTheme,
+  linkDefaultTheme,
+  listDefaultTheme,
+  menuDefaultTheme,
+  modalDefaultTheme,
+  selectDefaultTheme,
+  spinnerDefaultTheme,
+  switchDefaultTheme,
+  tableDefaultTheme,
+  tabsDefaultTheme,
+  tagDefaultTheme,
+}
+
+/**
+ * Allows for type safety groups of styles that will be merged.
+ *
+ * This is for merging default imported theming with custom styles.
+ *
+ * @param defaultTheming - The related object that comes from the Charka default themes
+ *
+ * @param styleObjs - The following style objects to be merged
+ */
+export function defineMergeStyles(
+  defaultTheming?: SystemStyleObject,
+  ...styleObjs: SystemStyleObject[] | undefined[]
+) {
+  return merge(defaultTheming, ...styleObjs)
+}
```

Comment:
This function was created for re-usability to handle merging of default styles with custom stylings. It uses the lodash `merge` function to handle recursive merging to prevent unexpected overrides.

Though this works as intended, I believe this could be improved, as there are multiple circumstances that this would be used in.

Ex. 1: Simple merge
```tsx
// From custom Link config
baseStyle: defineMergeStyles(linkDefaultTheme.baseStyle, {
    color: "primary",
    textDecoration: "underline",
    _focus: {
      boxShadow: "none",
    },
    _focusVisible: {
      boxShadow: "none",
      outline: "auto",
    },
  },
})
```

Ex. 2: Style function props needed to be passed to a default theme that uses some of those props (type error throws otherwise)
```tsx
// From custom Button config
const variantOutline = defineStyle((props) =>
  defineMergeStyles(defaultVariants?.outline(props), commonOutline)
)
```

Ex. 3: There may be a circumstance where a default baseStyle would be combined with a default variant because the component may not use other variants in the project, thereby compacting the custom config.
```tsx
const baseStyle = defineMergeStyles(
  defaultBaseStyle,
  defaultVariant.outline,
  {
    // Custom styles
  }
)
```

Reply:
@pettinarip 

> Awesome. Would love to have this documented somewhere...I feel bad to lose this doc in this PR.
I can put this up in the PR description as `Additional Information`

With your second suggestion, this approach will return a type equivalent only to the first object, which would be the default theme, and therefor lock down what the second object could have in its properties.

With the first suggestion, this will throw a type error in configs where the function is used in a `defineMultiStyleConfig` property.

For example, the `sizes` property in the [Button theme config](https://github.com/ethereum/ethereum-org-website/blob/1573a8c0b8646016a356c1113c992371cccfd3ba/src/%40chakra-ui/gatsby-plugin/components/Button.ts#L114-L118). The error thrown here:
```bash
Type 'SystemStyleObject' is not assignable to type 'Dict<SystemStyleInterpolation> | undefined'
```

`Dict` is currently a private type in Chakra, but it's simply
```ts
type Dict<T = any> = {
  [ x: string ]: T
}
```

We could apply this assertion on the return, written as `Record<string, SystemStyleObject>` which will work everywhere.
```ts
export function defineMergeStyles(
  defaultTheming?: SystemStyleObject,
  ...styleObjs: SystemStyleObject[] | undefined[]
) {
  return merge(defaultTheming, ...styleObjs) as Record<string, SystemStyleObject>
}
```